0000-route-broadcasting-protocol/0000-route-broadcasting-protocol.md

```diff
@@ -0,0 +1,226 @@
+
+![ILP Network](images/nodes.svg)
+
+Because a node has to route ILP packets to a certain next node appropriately depending on the destination address, so that the packets finally reach the specified address, the node needs to know which node can be the most efficient first hop for the destination ILP address. Therefore a node MUST know the mapping of:
+
+- Key
+  - `Prefix`: The ILP address prefix that this route can handle.
+- Value
+  - Route information
+    - `Next hop`: An identifier of the next node that is connected to the node directly.
+    - `Path`: How many hops the ILP packets will go through if sent. Generally, the number of elements of the path is expected to be less for efficiency because hops possibly cost network and server burden, or relaying fee.
+
+This mapping is called a **routing table**. In order to build a routing table, a node requests routing information to connected nodes respectively, or a node could possibly load static route configuration from a file.
+
+*A routing table may be a mere list that contains `Prefix`, `Next hop`, and `Path`, it doesn't need to be a map necessarily. That said, a map is more suitable for this case than a list because of the efficiency of looking up.*
+
+A node sends an ILP packet to the next node that is determined by referencing the routing table **of the sending node** and the further hops are NOT determined at that time. The next node determines, referencing **its own routing table**, which node could be the best when received the ILP packet. This chain of discrete decisions of routes is the basis of the routing of the Interledger network.
+
+### Routing Table Lifecycle
+The lifecycle of a routing table is like the below:
+
+- Created
+  - The table is empty at this point
+- Updated repeatedly
+  - By static configuration, though this is usually done only once when the node starts up
+  - By updates from the other nodes
+- Disposed
+  - When the node is shutdown
+  - When the node is restarted
+
+### Routing Table ID
+A routing table has its identifier, that is called **routing table ID**. A routing table ID is used for identifying its instance, and the ID is given when the table is created as empty. A routing table ID is a [UUID v4](https://tools.ietf.org/html/rfc4122#section-4.1.3).
+
+### Epoch
+Every time the routing table is modified, the update is logged and the revision number of the table is increased. The revision number of the table is called an **epoch**.
+
+A node maintains the epoch that is already known to the counter node, and considers it when requesting route updates so that the node can send only the difference (a stack of adding or withdrawing routes) from the last update with the last known epoch as the starting point. So the node doesn't need to send the whole routing table.
```

Comment:
If we use the term "counterpart node", let's use it here as well, instead of "counter node".

Reply:
Yes, the terminology should be unified, thank you.